lib/CodeGen/AsmPrinter/AsmPrinter.cpp

```diff
@@ -2542,6 +2542,22 @@ MCSymbol *AsmPrinter::GetExternalSymbolSymbol(StringRef Sym) const {
   return OutContext.getOrCreateSymbol(NameStr);
 }
 
+MCSymbol *AsmPrinter::GetSectionSymbol(const GlobalObject *GO) const {
+  MCSection *Sec = getObjFileLowering().SectionForGlobal(GO, TM, MMI);
+  return Sec->getBeginSymbol();
+}
+
+unsigned AsmPrinter::GetPGLTIndex(const GlobalObject *GO) {
+  const MCSection *Sec = getObjFileLowering().SectionForGlobal(GO, TM, MMI);
+  auto I = std::find(PGLT.begin(), PGLT.end(), Sec);
+  if (I != PGLT.end()) {
+    return (I - PGLT.begin()) + 1;
+  } else {
+    PGLT.push_back(Sec);
+    return PGLT.size();
```

Comment:
Yes, that was intentional. I'll comment it, but I think it's better this way than storing a temporary size and returning size+1.